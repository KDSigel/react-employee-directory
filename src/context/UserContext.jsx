import { createContext, useContext, useState } from 'react';
import { getUser } from '../services/users';

const UserContext = createContext();

const UserProvider = ({ children }) => {
  const currentUser = getUser();
  const [user, setUser] = useState(
    currentUser ? { id: currentUser.id, email: currentUser.email } : {}
  );

  return (
    <UserContext.Provider value={{ user, setUser }}>
      {children}
    </UserContext.Provider>
  );
};

const useUser = () => {
    const userCtx = useContext(UserContext)
    if (userCtx === undefined) {
        throw new Error('useUser must be used within a UserProvider');
    }
    return userCtx;
};

export { UserProvider, useUser };
