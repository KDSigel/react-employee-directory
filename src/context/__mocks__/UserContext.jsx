import { createContext, useContext, useState } from 'react';

const UserContext = createContext();

const UserProvider = ({ children, mockUser }) => {

  const [user, setUser] = useState(
    mockUser ? { id: mockUser.id, email: mockUser.email } : {}
  );

  return (
    <UserContext.Provider value={{ user, setUser }}>
      {children}
    </UserContext.Provider>
  );
};

const useUser = () => {
    const userCtx = useContext(UserContext)
    if (userCtx === undefined) {
        throw new Error('useUser must be used within a UserProvider');
    }
    return userCtx;
};

export { UserProvider, useUser };
