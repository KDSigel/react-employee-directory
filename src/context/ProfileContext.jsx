import { createContext, useContext, useState } from 'react';
import { getProfile } from '../services/profiles';
import { useUser } from './UserContext';


const ProfileContext = createContext();


const ProfileProvider = ({ children }) => {
  const {user} = useUser();
  const currentProfile = getProfile();
  const [profile, setProfile] = useState(
    currentProfile.name
      ? {
          name: currentProfile.name,
          email: user.email,
          bio: currentProfile.bio,
          birthday: currentProfile.birthday,
        }
      : {}
  );

  return (
    <ProfileContext.Provider value={{ profile, setProfile }}>
      {children}
    </ProfileContext.Provider>
  );
};

const useProfile = () => {
    const profileCtx = useContext(ProfileContext)
    if (profileCtx === undefined) {
        throw new Error('useProfile must be used within a ProfileProvider');
    }
    return profileCtx
};

export { ProfileProvider, useProfile };
