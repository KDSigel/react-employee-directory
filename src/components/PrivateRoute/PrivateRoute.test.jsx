import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { UserProvider } from '../../context/UserContext';
import PrivateRoute from './PrivateRoute';

jest.mock('../../context/UserContext')
jest.mock('../../services/users')

it('should render the PrivateRoute component', () => {
    const { container } = render(
        <UserProvider>
        <MemoryRouter> 
          <PrivateRoute location="/"/>
        </MemoryRouter>
        </UserProvider>
      );

      expect(container).toMatchSnapshot();
    });