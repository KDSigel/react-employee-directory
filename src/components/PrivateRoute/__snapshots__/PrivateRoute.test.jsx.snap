// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the PrivateRoute component 1`] = `<div />`;
